### ISSUE TITLE
Disable stream switching when self-hosting

### ISSUE THREAD

**Issue by michaelmsm89** [NancyFx/Nancy#1892]
I am self-hosting and I want to disable stream switching for request streams.

Looking at 

```C#
var nancyRequestStream = new RequestStream(owinRequestBody, ExpectedLength(owinRequestHeaders), false);
```

in Nancy.NancyMiddleware.cs, it looks as if false is always used.

It also seems to be hard-coded to true for ASP.NET hosted applications, with the following line from Nancy.Hosting.Aspnet.NancyHandler.cs

```C#
body = RequestStream.FromStream(context.Request.InputStream, expectedRequestLength, true);
```
Is there a way to configure this setting?

**Comment by khellang** [comment]
If you want a way, you could add a new property to the static `StaticConfiguration` class and use that for in the different hosts, then send a PR :wink: Since the different hosts have different default values, it makes sense to make the new property (something like `EnableRequestStreamSwitching`) a `Nullable<bool>` and do

    var nancyRequestStream = new RequestStream(owinRequestBody, ExpectedLength(owinRequestHeaders), StaticConfiguration.EnableRequestStreamSwitching ?? false);

@michaelmsm89 Are you up for it?

**Comment by jchannon** [comment]
I think this has been fixed in #2015
